### PR TITLE
OIIO and VDB Glf extensions as imaging plugins

### PR DESCRIPTION
**Pull request by ablev-dwa**
### Description of Change(s)
Refactors OIIO and VDB extensions to Glf as plugins instead of core Glf functionality. This refactor moves the code to these directories:
- pxr/imaging/plugin/glfOiio
- pxr/imaging/plugin/glfVdb

Adds a new class to Glf, GlfFieldTexture, so that Storm can interact with field textures in an abstract way without linking VDB directly.

This change allows us to remove OIIO and OpenVdb dependencies from imaging, and still have the functionality available as plugins.

### CODE REVIEW

**Comment by ablev-dwa** [comment]
I'm working on setting up a Windows build to resolve the Windows-only build errors. I'll update the PR with Windows fixes soon.

**Comment by drwave** [comment]
Wow - awesome!

**Comment by c64kernal** [comment]
Thanks so much @ablev-dwa -- we're scrambling with 20.05 release right now, but hopefully can take a look more closely after that's done. It looks awesome, thank you!

**Comment by jilliene** [comment]
Filed as internal issue #USD-5971

File: pxr/imaging/plugin/glfVdb/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    message(STATUS
+            "Skipping ${PXR_PACKAGE} because PXR_ENABLE_GL_SUPPORT is OFF")
+    return()
+endif()
+
+if (PXR_BUILD_OPENVDB_PLUGIN)
+    list(APPEND optionalPublicClasses vdbTextureData)
+    list(APPEND optionalLibs ${OPENEXR_Half_LIBRARY} ${OPENVDB_LIBRARY})
+    list(APPEND optionalIncludeDirs ${OPENVDB_INCLUDE_DIR})
+endif()
```
Comment (c64kernal) [inline pxr/imaging/plugin/glfVdb/CMakeLists.txt:14]: Hey @ablev-dwa -- was just looking at this to scope it out for size, and quickly noticed that maybe we don't need the above 5 lines?

**Comment by c64kernal** [comment]
@ablev-dwa We have a few changes coming up right after 20.05 is released that will cause a few more conflicts, so don't worry about resolving these right away so that you don't have to keep resolving them. Once we have the changes pushed, I'll let you know. (This change is awesome, but it's going to take some non-trivial amount of time to integrate it on our end, so thanks for your patience!)

File: pxr/imaging/plugin/glfVdb/CMakeLists.txt (hunk 1 above)
Comment (ablev-dwa) [reply pxr/imaging/plugin/glfVdb/CMakeLists.txt:14]: Whoops! You are correct, that slipped past me.

**Comment by ablev-dwa** [comment]
Thanks @c64kernal ! I'll wait for the all-clear and do a single update at that time.

**Comment by ablev-dwa** [comment]
@c64kernal Just checking in, are there more imaging changes on deck?

**Comment by c64kernal** [comment]
Hi @ablev-dwa -- apologies for the long wait. All of the big changes landed now, so we should be good to go. Thanks so much!

**Comment by ablev-dwa** [comment]
I'm closing this in favor of https://github.com/PixarAnimationStudios/USD/pull/1214 

The recent texturing changes don't seem to offer a clear way to abstract OpenVDB support from Storm and Glf in a way that was previously possible, and I don't have the time right now to develop a new approach. I hope to have time to revisit this later.

The new pull request separates OpenImageIO only into a standalone plugin, which is still useful.

**Comment by c64kernal** [comment]
Thanks @ablev-dwa and apologies for the trouble. Honestly we were going to split out the two anyway internally when we were going to merge this PR -- so this actually ends up being less work for us, but I'm sorry it's more complicated now for you :(

**Comment by ablev-dwa** [comment]
No trouble at all, I think the texture system overhaul is a big win for everyone even if it means a little more work on my end to realize this idea.

**Pull request closed without merging**